pkg/controller/envoy/ads_event.go

```diff
@@ -227,25 +237,25 @@ func (svc *ServiceEvent) handleRdsResponse(rsp *service_discovery_v3.DiscoveryRe
 		routeConfiguration = &config_route_v3.RouteConfiguration{}
 	)
 
+	current := sets.New[string]()
 	for _, resource := range rsp.GetResources() {
 		if err = anypb.UnmarshalTo(resource, routeConfiguration, proto.UnmarshalOptions{}); err != nil {
 			continue
 		}
-
 		apiStatus := core_v2.ApiStatus_UPDATE
-		newRdsResource := resource.String()
-		if newRdsResource != svc.DynamicLoader.RouteCache.GetRdsResource(routeConfiguration.GetName()) {
-			svc.DynamicLoader.RouteCache.SetRdsResource(routeConfiguration.GetName(), newRdsResource)
+		current.Insert(routeConfiguration.GetName())
+		newHash := hash.Sum64String(resource.String())
+		if newHash != svc.DynamicLoader.RouteCache.GetRdsHash(routeConfiguration.GetName()) {
+			svc.DynamicLoader.RouteCache.SetRdsHash(routeConfiguration.GetName(), newHash)
 			log.Debugf("[CreateApiRouteByRds] update %s", routeConfiguration.GetName())
 		} else {
 			apiStatus = core_v2.ApiStatus_UNCHANGED
 		}
-
 		svc.DynamicLoader.CreateApiRouteByRds(apiStatus, routeConfiguration)
 	}
 
```

Comment:
Doesn't rds need to "diff then remove" ?

Reply:
Missed that